frontend/src/metabase/dashboard/containers/DashboardApp.jsx

```diff
@@ -120,6 +133,15 @@ const DashboardApp = props => {
     onTimeout,
   });
 
+  const collectionId = dashboard?.collection_id;
+
+  useEffect(() => {
+    if (collectionId) {
+      setCollectionId(collectionId);
+      return () => clearBreadcrumbs();
+    }
```

Comment:
From my testing, this doesn't show the breadcrumb when the dashboard is on a `root` collection aka `Our analytics`. I don't think this is the intent as we also show the exact same breadcrumb for questions in the root collection.

This is how the question call `setCollectionId`

https://github.com/metabase/metabase/blob/70ba4d5c73022af1d5113a29429a3534dbfce67e/frontend/src/metabase/query_builder/containers/QueryBuilder.jsx#L261-L266

Not sure if we really need to check if `collectionId` exists, but remove the if clause makes the breadcrumb appear for dashboard on the root collection.

```suggestion
    setCollectionId(collectionId);
    return () => clearBreadcrumbs();
```